src/main/kotlin/lotto/model/LottoResult.kt

```diff
@@ -0,0 +1,30 @@
+package lotto.model
+
+const val MINIMUM_MATCHES = 3
+
+class LottoResult(private val winningNumbers: List<Int>, private val tickets: List<TicketModel>) {
+    fun calculateResult(): Map<Int, Int> {
+        return tickets.groupingBy { ticket ->
+            ticket.numbers.count { it in winningNumbers }
+        }
+            .eachCount()
+            .filterKeys { it >= MINIMUM_MATCHES }
+    }
+
+    fun totalPrize(): Int{
+        val prizeMap = mapOf(
+            3 to 5_000,
+            4 to 50_000,
+            5 to 1_500_000,
+            6 to 2_000_000_000
+        )
```

Comment:
Many functions, including `totalPrize()`, are only referenced in test packages, so they do not need to be opened as public functions.

Private functions separated only for readability reasons can also be considered verifiable by the public. Since the public function uses the private function, it can be naturally included in the test range.
However, if they are needed more than just improving readability, you might want to think about whether it's time to create another **object** that performs that role to the testable implementation.